google/cloud/spanner/samples/postgresql_samples.cc

```diff
@@ -207,6 +215,444 @@ void DmlGettingStartedUpdate(google::cloud::spanner::Client client) {
+  std::cout << "Update was successful.\n";
+}
+// [END spanner_postgresql_batch_dml]
+
+void DropTables(google::cloud::spanner_admin::DatabaseAdminClient client,
+                google::cloud::spanner::Database const& database) {
+  std::vector<std::string> statements = {
+      R"""(
+        DROP TABLE Albums
+      )""",
+      R"""(
+        DROP TABLE Singers
+      )""",
+  };
+  auto metadata =
+      client.UpdateDatabaseDdl(database.FullName(), statements).get();
+  if (!metadata) throw std::runtime_error(metadata.status().message());
+  std::cout << "Tables dropped.\nNew DDL:\n" << metadata->DebugString();
+}
+
+// [START spanner_postgresql_case_sensitivity]
+void CaseSensitivity(
+    google::cloud::spanner_admin::DatabaseAdminClient admin_client,
+    google::cloud::spanner::Database const& database,
+    google::cloud::spanner::Client client) {
+  std::vector<std::string> statements = {
+      R"""(
+        CREATE TABLE Singers (
+            -- SingerId will be folded to "singerid"
+            SingerId        BIGINT NOT NULL,
+            -- FirstName and LastName are double-quoted and will therefore
+            -- retain their mixed case and are case-sensitive. This means
+            -- that any statement that references any of these columns must
+            -- use double quotes.
+            "FirstName"     CHARACTER VARYING(1024),
+            "LastName"      CHARACTER VARYING(1024),
+            SingerInfo      BYTEA,
+            PRIMARY KEY(singerid)
+        )
+      )""",
+      R"""(
+        CREATE TABLE Albums (
+            SingerId        BIGINT NOT NULL,
+            AlbumId         BIGINT NOT NULL,
+            AlbumTitle      CHARACTER VARYING,
+            MarketingBudget BIGINT,
+            PRIMARY KEY(SingerId, AlbumId),
+            FOREIGN KEY(SingerId) REFERENCES Singers(SingerId)
+        )
+      )""",
+  };
+  auto metadata =
+      admin_client.UpdateDatabaseDdl(database.FullName(), statements).get();
+  if (!metadata) throw std::runtime_error(metadata.status().message());
+  std::cout << "Tables created.\nNew DDL:\n" << metadata->DebugString();
+
+  // Column names in mutations are always case-insensitive, regardless
+  // of whether the columns were double-quoted or not during creation.
+  auto insert_singers =  //
+      google::cloud::spanner::InsertMutationBuilder(
+          "Singers", {"singerid", "firstname", "lastname"})
+          .EmplaceRow(1, "Marc", "Richards")
+          .EmplaceRow(2, "Catalina", "Smith")
+          .Build();
+  auto commit =
+      client.Commit(google::cloud::spanner::Mutations{insert_singers});
+  if (!commit) throw std::runtime_error(commit.status().message());
+  std::cout << "Insert was successful.\n";
+
+  // DML statements must also follow the PostgreSQL case rules.
+  commit = client.Commit(
+      [&client](google::cloud::spanner::Transaction txn)
+          -> google::cloud::StatusOr<google::cloud::spanner::Mutations> {
```

Comment:
This got me thinking, maybe we should have overloads for `client.Commit()` that takes a `std::function<Status(google::cloud::spanner::Transaction txn)>`

Reply:
It was just a brainwave. SGTM.